spec/sec-class-definitions-patch.html

```diff
@@ -53,225 +47,334 @@ <h1>Static Semantics: Early Errors</h1>
+    <emu-grammar>
+      UnaryExpression : AwaitExpression
+    </emu-grammar>
     <emu-alg>
-      1. <ins>Return *false*.</ins>
+      1. Return *true*.
     </emu-alg>
-  </emu-clause>
 
-  <emu-clause id="sec-class-definitions-static-semantics-containsundefinedcontinuetarget">
-    <h1><ins>Static Semantics: ContainsUndefinedContinueTarget</ins></h1>
-    <p>With parameters _iterationSet_ and _labelSet_.</p>
-    <emu-grammar collapsed><ins>ClassStaticBlockStatementList: [emmpty]</ins></emu-grammar>
+    <emu-grammar>
+      FunctionDeclaration : `function` BindingIdentifier `(` FormalParameters `)` `{` FunctionBody `}`
+
+      FunctionDeclaration : `function` `(` FormalParameters `)` `{` FunctionBody `}`
+
+      FunctionExpression : `function` BindingIdentifier? `(` FormalParameters `)` `{` FunctionBody `}`
+    </emu-grammar>
     <emu-alg>
-      1. <ins>Return *false*.</ins>
+      1. Return *false*.
     </emu-alg>
-  </emu-clause>
 
-  <emu-clause id="sec-class-definitions-static-semantics-hasdirectsuper">
-    <h1><ins>Static Semantics: HasDirectSuper</ins></h1>
-    <emu-grammar collapsed><ins>ClassStaticBlock: `static` `{` ClassStaticBlockBody `}`</ins></emu-grammar>
+    <emu-grammar>
+      MethodDefinition :
+        ClassElementName `(` UniqueFormalParameters `)` `{` FunctionBody `}`
+        `get` ClassElementName `(` `)` `{` FunctionBody `}`
+        `set` ClassElementName `(` PropertySetParameterList `)` `{` FunctionBody `}`
+    </emu-grammar>
     <emu-alg>
-      1. <ins>Return |ClassStaticBlockBody| Contains |SuperCall|.</ins>
+      1. Return ContainsAwait of |ClassElementName|.
     </emu-alg>
-  </emu-clause>
 
-  <emu-clause id="sec-class-definitions-static-semantics-propname">
-    <h1>Static Semantics: PropName</h1>
-    <emu-grammar collapsed><ins>ClassElement : ClassStaticBlock</ins></emu-grammar>
+    <emu-grammar>
+      GeneratorMethod : `*` ClassElementName `(` UniqueFormalParameters `)` `{` GeneratorBody `}`
+    </emu-grammar>
     <emu-alg>
-      1. <ins>Return ~empty~.</ins>
+      1. Return ContainsAwait of |ClassElementName|.
     </emu-alg>
-  </emu-clause>
 
-  <emu-clause id="sec-class-definitions-static-semantics-LexicallyDeclaredNames">
-    <h1>Static Semantics: LexicallyDeclaredNames</h1>
-    <emu-grammar collapsed><ins>ClassBlockStatementList: [empty]</ins></emu-grammar>
+    <emu-grammar>
+      GeneratorDeclaration : `function` `*` BindingIdentifier `(` FormalParameters `)` `{` GeneratorBody `}`
+
+      GeneratorDeclaration : `function` `*` `(` FormalParameters `)` `{` GeneratorBody `}`
+
+      GeneratorExpression : `function` `*` BindingIdentifier? `(` FormalParameters `)` `{` GeneratorBody `}`
+    </emu-grammar>
     <emu-alg>
-      1. <ins>Return a new empty List.</ins>
+      1. Return *false*.
     </emu-alg>
-    <emu-grammar collapsed><ins>ClassBlockStatementList: StatementList</ins></emu-grammar>
+
+    <emu-grammar>
```

Comment:
This is the same behavior as `Contains` and `ContainsArguments`, `ContainsAwait` is merely a specialization of `Contains` since I need to catch both _AwaitExpression_ and `for await` (which doesn't have its own nonterminal). `ContainsArguments` doesn't contain this editorial text, but I can add it if the editors thinks its necessary.